src/run.rs

```diff
@@ -0,0 +1,35 @@
+use crate::config::Conf;
+use std::collections::HashSet;
+use std::path::PathBuf;
+
+use clap::{Parser, Subcommand};
+
+#[derive(Parser, Debug)]
+#[clap(version = env!("CARGO_PKG_VERSION"), author = "Trunk Technologies Inc.")]
+pub struct Cli {
+    #[command(subcommand)]
+    pub subcommand: Option<Subcommands>,
+
+    // #[arg(short, long, num_args = 1..)]
+    pub files: Vec<String>,
+
+    #[clap(long)]
+    #[arg(default_value_t = String::from("HEAD"))]
+    pub upstream: String,
+
+    #[clap(long)]
+    /// optional path to write results to
+    pub results: Option<String>,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Subcommands {
+    // print default config for toolbox
+    /// Generate default configuration content for toolbox
+    Genconfig {},
```

Comment:
nit: does this variant need a c-like structure? This can be a plain variant

```suggestion
    Genconfig,
```